cmd/util/cmd/checkpoint-collect-stats/account_stats.go

```diff
@@ -0,0 +1,108 @@
+package checkpoint_collect_stats
+
+import (
+	"cmp"
+	"slices"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/onflow/flow-go/fvm/systemcontracts"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type accountFormat uint8
+
+const (
+	accountFormatUnknown accountFormat = iota
+	accountFormatV1
+	accountFormatV2
+)
+
+func (format accountFormat) MarshalJSON() ([]byte, error) {
+	switch format {
+	case accountFormatV1:
+		return []byte("\"v1\""), nil
+
+	case accountFormatV2:
+		return []byte("\"v2\""), nil
+
+	default:
+		return []byte("\"unknown\""), nil
+	}
+}
+
+type AccountStats struct {
+	stats
+	FormatV1Count      int            `json:"account_format_v1_count"`
+	FormatV2Count      int            `json:"account_format_v2_count"`
+	AccountsInFormatV2 []string       `json:"accounts_in_format_v2,omitempty"`
+	ServiceAccount     *AccountInfo   `json:"service_account,omitempty"`
+	EVMAccount         *AccountInfo   `json:"evm_account,omitempty"`
+	TopN               []*AccountInfo `json:"largest_accounts"`
+}
+
+type AccountInfo struct {
+	Address      string        `json:"address"`
+	Format       accountFormat `json:"account_format"`
+	PayloadCount uint64        `json:"payload_count"`
+	PayloadSize  uint64        `json:"payload_size"`
+}
+
+func getAccountStatus(
+	chainID flow.ChainID,
+	accounts map[string]*AccountInfo,
+) AccountStats {
+	accountsSlice := make([]*AccountInfo, 0, len(accounts))
+	accountSizesSlice := make([]float64, 0, len(accounts))
+
+	var accountsInFormatV2 []string
+	var accountFormatV1Count, accountFormatV2Count int
+
+	for _, acct := range accounts {
+		accountsSlice = append(accountsSlice, acct)
+		accountSizesSlice = append(accountSizesSlice, float64(acct.PayloadSize))
```

Comment:
Maybe just use a `uint64` for the account sizes, unless there's a particular reason to use a `float64` instead

Reply:
> Maybe just use a uint64 for the account sizes, unless there's a particular reason to use a float64 instead

Yes, `[]float64` is needed here by `montanaflynn/stats` package.